feat: support for structured scaffold in create command (#970)
sparckles/Robyn: docs_src/src/pages/documentation/example_app/index.mdx

$ python -m robyn --create
```

You can choose to have a simple starter format or an opinionated scaffold ike so...

```bash
$ python -m robyn --create
? Directory Path: myproject
? Need Docker? (Y/N) Y
? Please choose if you'd like the scaffold to be a simple starter kit or an opinionated structure

>   Simple

Reviewer: let's call it barebones and recommended structured

Author: Oki

Author: updated

Reviewer: @ashupednekar , I don't see an update here

❯ Structured
```

This, would result in the following output if you choose scaffold type as `simple`

```bash
$ python3 -m robyn --create
? Directory Path: .
? Need Docker? (Y/N) Y

> ? Please choose if you'd like the scaffold to be a simple starter kit or an opinionated structure

Reviewer: ```suggestion
? Please choose if you would like the scaffold to be a barebones starter kit or a recommended structure
```

  />
</div>

If you choose to go with the structured scaffold, this is how your project will look like

> > note: at the moment, only no-db and sqlalchemy are supported here, you can always plug in other integrations as you see fit

Reviewer: why not the other ones? we support postgres/prisma/sqlalchemy/sqlite in the simple one

Author: prisma/nosql we can add in the future...

sqlite, MySQL, oracle are supported by sqlalchemy... maybe we can add commented envs and requirements

Author: added

Reviewer: > prisma/nosql we can add in the future...

let's ship a feature complete PR please 😄 

Also, we will have to update the documentation

```bash
├── adaptors
│   ├── __init__.py
│   ├── models
│   │   ├── __init__.py
│   │   └── user.py
│   ├── mutators
│   │   └── __init__.py
│   └── selectors
│       ├── __init__.py
│       └── misc.py
├── alembic.ini
├── api
│   ├── handlers
│   │   ├── __init__.py
│   │   ├── probes.py
│   │   └── sample.py
│   └── middlewares
│       └── __init__.py
├── conf.py
├── config.env
├── devops
│   ├── Dockerfile
│   ├── Dockerfile.src
│   └── docker-compose.yaml
├── migrations
│   ├── README
│   ├── env.py
│   ├── script.py.mako
│   └── versions
│       └── __init__.py
├── requirements.txt
├── server.py
└── utils
    ├── __init__.py
    └── db.py

12 directories, 24 files
```

Here's what each of these stand for

- server.py

This is where you instantiate your robyn server and inject global dependencies

```python
from robyn.helpers import discover_routes
from robyn import Robyn

from utils.db import get_pool
from conf import settings

> app: Robyn = discover_routes("api.handlers")

Reviewer: we don't have a discover_routes function

Author: It's added in helpers as part of this PR...

```python
def discover_routes(handler_path: str = "api.handlers") -> Robyn:
    mux: Robyn = Robyn(__file__)
    package = importlib.import_module(handler_path)
    for _, module_name, _ in pkgutil.iter_modules(package.__path__, package.__name__ + "."):
        module = importlib.import_module(module_name)
        logger.info(f"member: {module}")
        mux.include_router(module.router)
    return mux
```
